Adds toolbox (wide bar)

aburone opened the pull request:
Also trying teal for focus color on dark theme

aburone committed:
Adds toolbox (wide bar)
Tries teal for focus on dark theme

fvsch commented:
Thanks for the PR. I have a few requests.

1. Can we keep all example boxes the same width? I find that it's easier to read the page that way.
2. Can you remove the border from the `.toolbar` class, and back to the `.example-box` class? It messes up the box model (making the content area 22px instead of 24px).
3. The teal color for dark mode is interesting, but that's a far-reaching design change beyond the scope of this experiment. Can we keep shades of Photon Blue for now?

A few details:

- Currently, "active" toolbox buttons do not have a top border (unlike active tabs).
- You can position the `::before` top border using absolute position rather than negative margins, and give it a height:2px and background color (or a border-top) rather than an outline.

aburone committed:
Moves border back from .toolbar to .example box
Removes border from .toolbox .toolbar-button so hover and focus have
the same width

aburone committed:
Eliminates full width simulation
Moved back border from .toolbar to .example-box and gives toolbox a
border:0 so :hover stills taking full height
:focus color to blue
Removes .toolbox .active border from icons buttons with no text

aburone commented:
> Thanks for the PR. I have a few requests.
> 
>     1. Can we keep all example boxes the same width? I find that it's easier to read the page that way.

ok

> 
>     2. Can you remove the border from the `.toolbar` class, and back to the `.example-box` class? It messes up the box model (making the content area 22px instead of 24px).

I've originally moved it because it was interfering the the hover status of toolbox (with has to cover the entire height). Found a way around, done

> 
>     3. The teal color for dark mode is interesting, but that's a far-reaching design change beyond the scope of this experiment. Can we keep shades of Photon Blue for now?

It was just a test. back to blue. used one without alpha in dark, now it's visible.

> 
> 
> A few details:
> 
>     * Currently, "active" toolbox buttons do not have a top border (unlike active tabs).

Done, added a differentiation

> 
>     * You can position the `::before` top border using absolute position rather than negative margins, and give it a height:2px and background color (or a border-top) rather than an outline.

That one is hard. In the current codebase it's it's a div added in the middle. That's with it doesn't move the button.

Having :focus as an inset box-shadow, if I use a border the box-shadow will not take the whole height of the toolbar
And if I use absolute positioning the width is missing.
I'm trying other options...


BTW: shouldn't we use nav instead of divs so it's easier for screen readers? (right now I can't find a way to move to devtools with the keyboard.

aburone committed:
Uses postion:relative instead of margins for toolbox :active top line
Adds top line on hover
Removes redundant rules
